### DISCUSSION TITLE
Smartthing e caddy

### DISCUSSION THREAD

**Discussion by ronyn77** [indomus/forum#2307] [HUB: Home Assistant]
Salve, 

ho installato caddy e funziona tranquillamente il reverse proxy. Non ho nessun problema se non con l'integrazione Smartthing. Quando provo ad inserire il token mi dice "Impossibile configurare SmartApp. Riprova".
Noto che nell'indirizzo del webhook, mette l'indirizzo interno della lan e non quello definito con l'https con l'appropriato dns. Quindi se clicco sul webhook, la pagina non carica. Se modifico il link e metto l'appropriato dns, mi carica  una pagina con l'errore 405: Method Not Allowed.
Ho guardato altre guide per aggirare il problema ma con Nginx, dove venivano aggiunte delle righe per caricare i certificati nel file configuration.yaml. Ma per la configurazione con caddy non so che fare.

Quale idea?

**Comment by snakuzzo** [comment]
Premesso che non ho nulla da integrare con Smarthing, ma tutto quello che hai detto non c'entra niente con il tuo problema. NGINX e/o Caddy non c'entrano assolutamente nulla.
Il tuo problema non sono i certificati...e te lo dice pure nel messaggio di errore quando provi a forzare il nominio.
`405 - Method Not Allowed` vuol dire che non puoi fare quella chiamata in `GET` così come stai facendo.

Devi seguire le indicazioni riportate sulla documentazione ufficiale [QUI](https://www.home-assistant.io/integrations/smartthings/#webhook). Credo che il tuo problema sia legato all'impostazione corretta di external ed internal URL, che dovrebbero essere impostati come di seguito:

es.

```
internal_url: http://192.168.1.10:8123
external_url: https://tuodominio.duckdns.org:9000
```

**Reply by ronyn77** [reply]
Ascoltami...quando ho installato mesi fa l'integrazione di smartthing, purtroppo ci capivo poco di home assistant, e di conseguenza l'ho impostato male. Adesso sto provando a sistemare le cose, ma sfortunatamente non riesco a cancellare l'integrazione e quindi sto provando a cambiare le cose in modo manuale scrivendo direttamente sui file nelle cartelle di installazione di home assistant. Ovviamente cio' sta portando ad altri errori.
Quello che vorrei fare e cancellare l'integrazione se possibile, per poi reinstallarla inserendo i dati corretti. L'integrazione non compare visualmente, ma quando provo ad aggiungerla mi genera sempre il webhook vecchio. Voglio rimuovere le vecchie informazioni, ma non riesco.
Ho letto che posso usare l'applicativo hass_smartthings_remove, ma non ho capito come farlo funzionare.

Hai idee su come posso fare?

**Reply by snakuzzo** [reply]
Ma perché devi rimuovere e rifare da capo.
Ti ho detto qual è il problema, poi fai tu.

Non conosco né l'integrazione, né questo tool per la rimozione, ma comunque basta cercare su Google ed il primo risultato ti dice tutto

https://pypi.org/project/hass-smartthings-remove/

Se usi Docker devi prima lanciare una shell nel container e poi eseguire quel comando.
Es.

```
docker exec -it homeassistant bash
``` 

poi installi il tool

```
pip install hass_smartthings_remove
```

e poi lo esegui

```
hass_smartthings_remove PERSONAL_ACCESS_TOKEN
```

**Reply by ronyn77** [reply]
>internal_url: http://192.168.1.10:8123
>external_url: https://tuodominio.duckdns.org:9000

Non erano proprio impostati nel mio yaml, effettivamente non l'avevo letta quella parte della documentazione e stavo cercando soluzioni piu complicate. Grazie.
In ogni caso, non ha funzionato lo stesso. Dopo aver inserito il token, mi ha fatto loggare nel sito della Samsung ed ha generato l'errore "ERROR: Something went wrong. Please try to install the SmartApp again".
>https://pypi.org/project/hass-smartthings-remove/

L'avevo già trovata questa applicazione, ma non ho capito come fare ad installarla.
Non ho home assistant su docker. Ho la versione supervised credo, comunque ho seguito questa procedura : https://www.home-assistant.io/installation/generic-x86-64

Ho fatto un paio di cose, ma non ne vengo a capo. Ho installato ubuntu sotto windows, ed ho installato il programma qui sopra menzionato, ma non so come poi applicarlo "dentro" home assistant.
Quindi poi ho guardato su internet come fare con SSH. Quindi mi sono connesso con putty, ho installato l'ambiente python, e qui non riesco ad andare avanti per ora.

**Reply by snakuzzo** [reply]
Aspetta, ma questa applicazione rimuove qualcosa dal cloud di SmartThing, non da HA.
Allora puoi eseguire quegli step da qualsiasi PC abbia python, non devi farlo dal container di HA. Non serve installare Ubuntu sotto Windows. Python c'è anche su Windows e su altri OS. Puoi farlo da dove ti pare.

Se non sei pratico, installa l'ultima release di Python su Windows (cerca su Google come fare...devi scaricare un pacchetto ed installarlo, niente di che) e poi esegui quei due comandi ... sempre da riga di comando di Windows

**Reply by ronyn77** [reply]
No, non sono pratico di queste cose, effettivamente...per quello sto riscontrando difficoltà....avevo installato il Python dentro 'ha' facendo un ssh da putty...probabilmente ho fatto una boiata...
...quindi l'applicazione toglie dal cloud e non da 'ha'? Io pensavo il contrario...infatti facevo di tutto per cercare di installarlo su ha o di trovare un modo per farlo comunicare con 'ha'...

**Reply by snakuzzo** [reply]
sì, confermo che hai fatto una boiata, anche perché Python già c'è...HA è scritto in python...quindi...fai te :)

Da quello che ho capito fa quello...ma non ci ho perso più di 30s...non è documentato nulla.

**Reply by ronyn77** [reply]
Be, allora grazie.
Più tardi provo e vediamo come va a finire.
